packages/mco/components/disaster-recovery/applications/failover-and-relocate-modal/subscription-group-selector.tsx

```diff
@@ -0,0 +1,213 @@
+    const memoizedAcmToDRState = useDeepCompareMemoize(
+      state.acmToDRState,
+      true
+    );
+    // To avoid re-rendering when a new selectedTargetCluster object  is generated with same props
+    const memoizedTargetCluster = useDeepCompareMemoize(
+      state.selectedTargetCluster,
+      true
+    );
+    // To avoid re-rendering when a new selectedDRPolicy object is generated with same props
+    const memoizedAcmToDRPolicy = useDeepCompareMemoize(
+      state.selectedDRPolicy,
+      true
+    );
+    const [isOpen, setOpen] = React.useState(false);
+    const [options, setOptions] = React.useState([]);
+
+    const setSelected = React.useCallback(
+      (selection: string[], isUpdate?: boolean) => {
+        dispatch({
+          type: FailoverAndRelocateType.SET_SELECTED_SUBS_GROUP,
+          payload: {
+            selected: selection,
+            isUpdate,
+          },
+        });
+      },
+      [dispatch]
+    );
+
+    const setErrorMessage = React.useCallback(
+      (errorMessage: string) => {
+        dispatch({
+          type: FailoverAndRelocateType.SET_ERROR_MESSAGE,
+          payload: {
+            subscriptionGroupErrorMessage: errorMessage,
+          },
+        });
+      },
+      [dispatch]
+    );
+
+    React.useEffect(() => {
+      // Filter all DRPlacements under a selected target cluster
+      if (
+        !!Object.keys(memoizedTargetCluster).length &&
+        memoizedTargetCluster?.isClusterAvailable
+      ) {
+        const validState: string[] = [];
+        const generatedOptions = memoizedAcmToDRState.reduce(
+          (arr, acmToDRState, index) => {
+            if (
+              validateDRPolicy(acmToDRState, memoizedAcmToDRPolicy?.policyName)
+            ) {
+              const isValidTargetCluster = validateTargetCluster(
+                acmToDRState,
+                memoizedTargetCluster?.clusterName
+              );
+              isValidTargetCluster &&
+                validState.push(getName(acmToDRState.drPolicyControl));
+              return [
+                ...arr,
+                getOptions(acmToDRState, index, isValidTargetCluster, t),
+              ];
+            } else {
+              return arr;
+            }
+          },
+          []
+        );
+        if (!!generatedOptions.length) {
+          setSelected(validState);
+          setOptions(generatedOptions);
+          setErrorMessage('');
+        } else {
+          setOptions([]);
+          setSelected([]);
+          setErrorMessage(t('No subscription groups are found'));
+        }
+      } else {
+        setSelected([]);
+        setOptions([]);
+      }
+    }, [
+      memoizedAcmToDRState,
+      memoizedAcmToDRPolicy,
+      memoizedTargetCluster,
+      dispatch,
+      setSelected,
+      setErrorMessage,
+      setOptions,
+      t,
+    ]);
+
+    const onToggle = (isOpen) => {
+      setOpen(isOpen);
```

Comment:
on toggling state should negate right ?
```suggestion
      setOpen(!isOpen);
```

Reply:
the same here, selector will pass this as arg, we are using  state variable